osdep/WindowsEthernetTap.cpp

```diff
@@ -1250,6 +1256,88 @@ void WindowsEthernetTap::_setRegistryIPv4Value(const char *regKey,const std::vec
 	}
 }
 
+void WindowsEthernetTap::_allowICMPv4(std::string address)
+{
+	// allows icmp, scoped to a specific ip address and interface name
+
+	char nwString[32] = { 0 };
+	sprintf(nwString, "%.16llx", _nwid);
+	std::string nwString2 = { nwString };
+
+		std::string cmd = R"(C:\Windows\System32\WindowsPowershell\v1.0\powershell.exe "New-NetFirewallRule -DisplayName zerotier-icmpv4-)" + nwString2 + address +
+			R"( -InterfaceAlias 'ZeroTier One `[)" + nwString2 + R"(`]')" + 
+			" -Protocol ICMPv4 -Action Allow" + 
+			" -LocalAddress " + address + "\"\r\n";
+
+		fprintf(stderr, cmd.c_str());
+
+		STARTUPINFOA startupInfo;
+		PROCESS_INFORMATION processInfo;
+		startupInfo.cb = sizeof(startupInfo);
+		memset(&startupInfo, 0, sizeof(STARTUPINFOA));
+		memset(&processInfo, 0, sizeof(PROCESS_INFORMATION));
+
+		if (CreateProcessA(NULL, (LPSTR)cmd.c_str(), NULL, NULL, FALSE, CREATE_NO_WINDOW, NULL, NULL, &startupInfo, &processInfo)) {
+		WaitForSingleObject(processInfo.hProcess, INFINITE);
+
+		CloseHandle(processInfo.hProcess);
+		CloseHandle(processInfo.hThread);
+		}
+}
+
+void WindowsEthernetTap::_allowICMPv4_cleanup(std::string addr)
+{
+		// removes 1 icmp firewall rule
+
+		char nwString[32] = { 0 };
+		sprintf(nwString, "%.16llx", _nwid);
```

Comment:
The format string is probably fine to leave as-is because this is very platform-specific code, but I try to use the ugly PRIx64 macro from <cinttypes> to be completely general, e.g., if this were to ever exist on a platform where long long is not 64-bits:
```
#include <cinttypes>
...
sprintf(nwString, "%.16" PRIx64, _nwid);
```

Reply:
Thanks. I have only a foggy idea of how strings work and that section specifically was annoying me.